tiledb/sm/array/array.cc

```diff
@@ -109,6 +111,11 @@ const URI& Array::array_uri() const {
   return array_uri_;
 }
 
+const URI& Array::array_uri_serialized() const {
+  std::unique_lock<std::mutex> lck(mtx_);
+  return array_uri_serialized_;
```

Comment:
While this was in the original change, it seems odd to me: we’re locking a mutex in order to return a reference which points to the `Array`’s `array_uri_serialized_`. But now the caller has an unprotected pointer into memory that should be controlled by a mutex. This means that if another thread calls `set_uri_serialized`, the holder of this pointer is subject to their `URI` changing out from underneath them, and all the delights and undefined behavior that brings with it.

Am I missing something?

Since this is a simple cherrypick, I don’t necessarily think this should *block* merging this, but it seems like it is worth addressing.

Reply:
@thetorpedodog there are larger locking changes in flight to adjust items like this. For now I kept it the same as the `array_uri` to simplify the next batch of changes around Array.